packages/static/src/progress-bar/progress-bar.tsx

```diff
@@ -0,0 +1,164 @@
+      color: inputColor = 'blue',
+      variant = 'normal',
+      value: inputValue = 0,
+      minValue: inputMinValue = 0,
+      maxValue: inputMaxValue = 100,
+      onComplete,
+      // html props
+      ...htmlProps
+    } = props
+
+    const color = useMemo(() => {
+      // support the old color variants
+      if (inputColor === 'blue') {
+        return '#009EDA'
+      }
+
+      if (inputColor === 'green') {
+        return '#107C10'
+      }
+
+      return inputColor
+    }, [inputColor])
+
+    const minValue = inputMinValue < 0 ? 0 : inputMinValue
+    const maxValue = inputMaxValue < minValue ? minValue : inputMaxValue
+    const value = clampValue(inputValue, minValue, maxValue)
+
+    const linkRef = useDOMRef<HTMLDivElement>(ref)
+    const labelProps = {
+      role: 'progressbar',
+      'aria-label': label ?? '',
+      'aria-valuemin': minValue,
+      'aria-valuenow': value,
+      'aria-valuemax': maxValue,
+    }
+
+    const progressPercentage = useMemo(() => {
+      if (maxValue - minValue === 0 || value === maxValue) {
+        return 100
+      }
+
+      const percentage = (value - minValue) / (maxValue - minValue)
+      return Math.round(percentage * 100)
+    }, [value, minValue, maxValue])
+
+    useEffect(() => {
+      if (progressPercentage === 100) {
+        onComplete?.()
+      }
+    }, [progressPercentage, onComplete])
+
+    return (
+      <CssInjection css={css} childrenRef={linkRef}>
+        <div
+          className={`cdg-progress-bar-wrapper ${styles.progressBarWrapper} ${className}`}
+          ref={linkRef}
+          {...labelProps}
+          {...htmlProps}
+        >
+          {(!!label || !!rightLabel) && (
+            <div
+              className={`cdg-progress-bar-label ${styles.progressBarLabel}`}
+            >
+              <span {...labelProps} className={styles.progressBarLabelSpan}>{label}</span>
+              {rightLabel && <span {...labelProps} className={styles.progressBarLabelSpan}>{rightLabel}</span>}
+            </div>
+          )}
+
+          <div
+            className={`cdg-progress-bar ${styles.progressBar} ${styles[
+              `variant${variant.charAt(0).toUpperCase() + variant.slice(1)}`
+            ]
+              } ${styles[`size${size.charAt(0).toUpperCase() + size.slice(1)}`]}`}
+            style={{ backgroundColor: barColor }}
+          >
+            {!loading && (
+              <div
+                className={`progress-bar__current-state ${styles.currentProgress}`}
+                style={{
+                  width: `${progressPercentage}%`,
+                  backgroundColor: color,
+                }}
+              />
+            )}
+
+            <div
+              className={`progress-bar__loading ${styles.loadingProgress} ${styles[
+                `loading${loading.toString().charAt(0).toUpperCase() +
+                loading.toString().slice(1)
+                }`
+              ]
+                }`}
+              style={{
```

Comment:
change it to prop css so it is easier for users to override. just like how it is done in tabs

Reply:
fixed